providers/FoliaProvider/src/main/java/com/palmergames/bukkit/towny/scheduling/impl/FoliaTaskScheduler.java

```diff
@@ -50,7 +50,7 @@ public boolean isRegionThread(final Location location) {
 
 	@Override
 	public ScheduledTask run(final Consumer<ScheduledTask> task) {
-		return runAsync(task);
+		return runGlobal(task);
```

Comment:
Why?

Reply:
Asynchronous threads in the Folia server are very limited. For example, an error will be thrown when using asynchronous call events. Generally, global synchronization threads are used.

Reply:
If you want to use asynchronous, you can use RunAsync(). Using asynchronous by default will cause some strange problems.

Reply:
Only things that need to run on the global scheduler should run there, in all other effects it's the same as running it async on folia (since the global scheduler is not sync to any region thread)